src/renderers/shared/utils/__tests__/ReactErrorUtils-test.js

```diff
@@ -63,13 +64,23 @@ describe('ReactErrorUtils', () => {
       global.process = {
         env: Object.assign({}, process.env, {NODE_ENV: 'production'}),
       };
-      jest.resetModuleRegistry();
+      jest.resetModules();
       ReactErrorUtils = require('ReactErrorUtils');
       expect(ReactErrorUtils.invokeGuardedCallback).toEqual(
         ReactErrorUtils.invokeGuardedCallbackWithCatch
       );
       __DEV__ = true;
       global.process = oldProcess;
     });
+
+    it('shouldn\'t call the callback again while executing synchronously induced other call in development', () => {
+      var callback = jest.fn(() => {
+        ReactErrorUtils.invokeGuardedCallback('foo', callback2, 'arg', true);
+      });
+      var callback2 = jest.fn();
+      ReactErrorUtils.invokeGuardedCallback('foo', callback, 'arg', true);
+      expect(callback).toHaveBeenCalledTimes(1);
+      expect(callback2).toHaveBeenCalledTimes(1);
+    });
```

Comment:
Yes. But test file would need to stay as in my PR so the ordering issue is solved by `resetModules`